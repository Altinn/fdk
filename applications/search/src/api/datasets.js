import axios from 'axios';
import qs from 'qs';

const datasetsUrlBase = '/datasets';

const searchAggregations =
  'accessRightsCount,theme_count,orgPath,catalogs,provenanceCount,firstHarvested,missingFirstHarvested,lastChanged,missingLastChanged,spatial,opendata';

export const datasetsSearchUrl = query =>
  `${datasetsUrlBase}${qs.stringify(
    { ...query, aggregations: searchAggregations },
    { addQueryPrefix: true }
  )}`;

export const getDataset = id =>
  axios
    .get(`${datasetsUrlBase}/${id}`)
    .then(response => response.data)
    .catch(e => console.log(JSON.stringify(e))); // eslint-disable-line no-console

export const getDatasetByURI = uri =>
  axios
    .get(`${datasetsUrlBase}/byuri?uri=${uri}`)
    .then(response => response.data)
    .catch(e => console.log(JSON.stringify(e))); // eslint-disable-line no-console
