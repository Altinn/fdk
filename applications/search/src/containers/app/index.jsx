import * as React from 'react';
import PropTypes from 'prop-types';
import qs from 'qs';
import { DropdownButton, MenuItem } from 'react-bootstrap';
import { browserHistory, Link } from 'react-router';

import localization from '../../components/localization';
import { addOrReplaceParam } from '../../utils/addOrReplaceUrlParam';
import './index.scss';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedLanguage: `${localization.lang['norwegian-nb']}`,
      selectedLanguageCode: 'nb'
    };
    this.onChangeLanguage = this.onChangeLanguage.bind(this);
    this.getLangUrl = this.getLangUrl.bind(this);
  }

  onChangeLanguage(e) {
    const langCode = e;
    const langUrl = this.getLangUrl(langCode);
    const nextUrl = `${location.pathname}${langUrl}`;
    browserHistory.push(nextUrl);

    let text;
    if (langCode === 'nb') {
      text = `${localization.lang['norwegian-nb']}`;
    } else if (langCode === 'nn') {
      text = `${localization.lang['norwegian-nn']}`;
    } else if (langCode === 'en') {
      text = `${localization.lang['english-en']}`;
    }
    this.setState({
      selectedLanguage: `${text}`,
      selectedLanguageCode: `${langCode}`
    });
    localization.setLanguage(langCode);
  }

  getLangUrl(langCode) {
    const href = window.location.search;
    const queryObj = qs.parse(window.location.search.substr(1));
    if (langCode === 'nb') {
      return addOrReplaceParam(href, 'lang', '');
    } else if (href.indexOf('lang=') === -1) {
      return href.indexOf('?') === -1 ? `${href}?lang=${langCode}` : `${href}&lang=${langCode}`;
    } else if (langCode !== queryObj.lang) {
      const replacedUrl = addOrReplaceParam(href, 'lang', langCode);
      return replacedUrl.substring(replacedUrl.indexOf('?'));
    }
    return href;
  }

  render() {
    // let queryObj = qs.parse(window.location.search.substr(1));
    // let language = queryObj.lang ? queryObj.lang : 'nb';
    const childWithProp =
      React.Children.map(this.props.children, child => React.cloneElement(child, {
        selectedLanguageCode: this.state.selectedLanguageCode
      }));
    return (
      <div>
        <div className="fdk-header-beta">
          {localization.beta.first}
          <a className="white-link" href="mailto:fellesdatakatalog@brreg.no">{localization.beta.second}</a> {localization.beta.last}
        </div>
        <div className="fdk-header">
          <div className="container">
            <div className="row">

              <div className="col-md-4">
                <Link to="/">
                  <img className="fdk-logo" src="/static/img/fdk-logo@2x.png" alt="Logo for Felles datakatalog" />
                </Link>
              </div>

              <div className="col-md-4">
                <p className="fdk-p-header-sub">
                  {localization.app.titleSub} {localization.app.readMore}
                  <a href="/about">
                    {localization.app.title}
                  </a>
                </p>
              </div>

              <div className="col-md-4 fdk-header-right">
                <div className="fdk-float-right">
                  <DropdownButton
                    id="search-menu-dropdown-1"
                    bsStyle="default"
                    className="dropdown-toggle fdk-button fdk-button-default fdk-button-on-white fdk-button-menu"
                    title={localization.app.menu}
                  >
                    <MenuItem
                      key="menu-1"
                      eventKey="menu-1"
                      href="/about">Om Felles Datakatalog</MenuItem>
                  </DropdownButton>
                </div>
                <div className="fdk-header-padding">
                  <div className="fdk-float-right fdk-margin-right-double">
                    <DropdownButton
                      id="search-language-dropdown-1"
                      bsStyle="default"
                      className="dropdown-toggle fdk-button-language"
                      title={localization.lang.chosenLanguage}
                      onSelect={this.onChangeLanguage}
                    >
                      <MenuItem key="1" eventKey="nb">{localization.lang['norwegian-nb']}</MenuItem>
                      <MenuItem key="2" eventKey="nn">{localization.lang['norwegian-nn']}</MenuItem>
                      <MenuItem key="3" eventKey="en">{localization.lang['english-en']}</MenuItem>
                    </DropdownButton>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        {childWithProp}
        <div className="fdk-footer">
          <div className="container">
            <div className="row">
              <div className="col-md-3">
                <p className="fdk-p-footer">
                  <a
                    href="https://www.brreg.no/personvernerklaering/"
                  >
                    {localization.footer.information}<br />
                    {localization.footer.privacy}
                    <i className="fa fa-external-link fdk-fa-right" />
                  </a>
                </p>
              </div>
              <div className="col-md-6 text-center">
                <p className="fdk-p-footer">
                  {localization.footer.information_text}
                </p>
              </div>
              <div className="col-md-3 text-right">
                <p className="fdk-p-footer">
                  <a
                    href="mailto:felleskatalog@brreg.no"
                  >
                    <div>{localization.footer.contact}</div>
                    {localization.footer.mail}
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  children: PropTypes.node.isRequired
};

