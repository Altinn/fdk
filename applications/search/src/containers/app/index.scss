@import '../../assets/css/bootstrap.min.css';
@import '../../assets/css/bootstrap-theme.min.css';
@import '../../assets/css/designsystem.scss';
@import '../../assets/css/colors.scss';
@import '../../assets/css/register.scss';
@import '../../assets/css/portal.scss';
@import '../../assets/css/typo.scss';
@import '../../assets/css/common.scss';
@import '../../assets/css/font-awesome.min.css';
//@import './assets/css/main.css';

$sk-bold-weight: 700 !default;
$sk-medium-weight: 500 !default;
$sk-light-weight: 300 !default;

$sk-font-family: -apple-system, BlinkMacSystemFont,
  "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell",
  "Fira Sans", "Droid Sans", "Helvetica Neue",
  sans-serif !default;

$sk-layout-background: #f4f4f4 !default;
$sk-layout-panel-background: white !default;
$sk-primary-theme-color: #bce6fa !default;
$sk-active-bg-color: #bce6fa !default;
$sk-border-color: #ccc !default;
$sk-title-color: black !default;
$sk-text-color: #444 !default;
$sk-text-secondary-color: #777 !default;
$sk-list-text-color: #666 !default;
$sk-list-secondary-text-color: #ccc !default;
$sk-action-text-color: #08c !default;
$sk-action-active-color: #000 !default;
$sk-disabled-text-color: #ddd !default;
$sk-remove-action-color: #fa2a00 !default;

$sk-selector-background: white !default;
$sk-selector-active-background: #08c !default;
$sk-selector-border-color: #ccc !default;
$sk-selector-text-color: black !default;
$sk-selector-active-text-color: white !default;

$sk-selected-filters-background: #f4f4f4 !default;

$sk-heading-font-size: 14px !default;
$sk-text-font-size: 14px !default;
$sk-list-font-size: 14px !default;
$sk-action-font-size: 14px !default;
$sk-list-secondary-font-size: 12px !default;

$sk-border-radius: 3px !default;

.publisher {
    position: absolute;
    top: 0;
    right: 0;
}

.container {
  position: relative;
}


.btn-group {
  .fdk-button-language {
    background-color: transparent !important;
    border-color: transparent;
    text-shadow: none;
    box-shadow: none;
    background-image: none;
    text-decoration: underline;
    &:focus, &:active, &:hover {
      background-color: transparent !important;
      border-color: transparent !important;
    }
  }
}

.fdk-header-padding {
  padding-top: 5px;
}

.fdk-button-menu {
  &:before {
    display: inline-block;
    font: normal normal normal 14px/1 FontAwesome;
    font-size: inherit;
    text-rendering: auto;
    -webkit-font-smoothing: antialiased;
    content: "\f0c9";
    margin-right: .2em;
  }
}

.mt-60 {
  margin-top: 4em;
}

.block {
  display: block;
}

/*
 TOP-HEADER
 */

.btn-group-default .dropdown-menu {
  left: auto;
  right: 0;
}

/*
SEARCHPAGE SORTING DROPDOWN
*/

.btn-group {
  .dropdown-toggle {
    &:focus, &:active, &:hover {
      background-color: $color-cta4;
      border-color: transparent;
      box-shadow: none;
    }
  }
}

.open>.dropdown-toggle.btn-default {
  background-color: $color-cta4;
  border-color: transparent;
  box-shadow: none;
}

/*
 FILTER CHECKBOX
 */
.checkbox-replacement {
  height: 16px;
  width: 16px;
  border: 1px solid #2D3741;
  border-radius: 3px;
}
.checkbox label {
  padding-left:0;
}
.checkbox label label {
  margin-right: 4px;
  vertical-align: middle;
  background: #fff;
  min-height: auto;
}
input:checked ~ .checkbox-replacement::after {
  position: absolute;
  width: 10px;
  height: 10px;
  background: #2D3741;
  left: 3px;
  top: 6px;
  border-radius: 3px;
  content:'';
}
.checkbox label input[type="checkbox"]{
  position: absolute;
  left:-9999px;
}
.sk-hits-grid {
  margin-top: 7px;
}

.fdk-button-language:focus {
	outline: auto !important;
}
