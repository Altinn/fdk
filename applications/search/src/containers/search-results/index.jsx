import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { browserHistory } from 'react-router';
import qs from 'qs';
import queryString from 'query-string';
import { Link, Route, Switch } from 'react-router-dom';

import localization from '../../components/localization';
import {
  fetchDatasetsIfNeeded,
  fetchTermsIfNeeded,
  fetchThemesIfNeeded
} from '../../actions/index';
import { addOrReplaceParam, getParamFromUrl, removeParam } from '../../utils/addOrReplaceUrlParam';
import ResultsDataset from '../../components/search-results-dataset';
import ResultsConcepts from '../../components/search-concepts-results';
import SearchBox from '../../components/search-app-searchbox';
import ResultsTabs from '../../components/search-results-tabs';
import { removeValue, addValue } from '../../utils/stringUtils';
import './index.scss';
// import '../../components/search-results-searchbox/index.scss';

const ReactGA = require('react-ga');
const sa = require('superagent');

const getTabUrl = (tab) => {
  const href = window.location.search;
  const queryObj = qs.parse(window.location.search.substr(1));
  if (href.indexOf('tab=') === -1) {
    return href.indexOf('?') === -1 ? `${href}?tab=${tab}` : `${href}&tab=${tab}`;
  } else if (tab !== queryObj.tab) {
    const replacedUrl = addOrReplaceParam(href, 'tab', tab);
    return replacedUrl.substring(replacedUrl.indexOf('?'));
  }
  return href;
}

class SearchPage extends React.Component {
  constructor(props) {
    super(props);
    const searchQuery = queryString.parse(props.location.search) || {
        searchQuery: {
          size: 50
        },
        showFilterModal: false,
    }

    this.state = {
      showConcepts: false,
      searchQuery
    }
    this.queryObj = qs.parse(window.location.search.substr(1));
    if (!window.themes) {
      window.themes = [];

      sa.get('/reference-data/themes')
        .end((err, res) => {
          if (!err && res) {
            res.body.forEach((hit) => {
              const obj = {};
              obj[hit.code] = {};
              obj[hit.code].nb = hit.title.nb;
              obj[hit.code].nn = hit.title.nb;
              obj[hit.code].en = hit.title.en;
              window.themes.push(obj);
            });
          }
        });
    }
    this.handleSelectView = this.handleSelectView.bind(this);
    this.handleHistoryListen = this.handleHistoryListen.bind(this);
    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
    this.handleSearchChange = this.handleSearchChange.bind(this);
    this.handleDatasetFilterThemes = this.handleDatasetFilterThemes.bind(this);
    this.handleDatasetFilterAccessRights = this.handleDatasetFilterAccessRights.bind(this);
    this.handleDatasetFilterPublisher = this.handleDatasetFilterPublisher.bind(this);
    this.handleDatasetSort = this.handleDatasetSort.bind(this);
    this.close = this.close.bind(this);
    this.open = this.open.bind(this);

    const datasetsURL = '/datasets';
    this.props.dispatch(fetchDatasetsIfNeeded(`/datasets/${props.location.search}`));
    this.props.dispatch(fetchTermsIfNeeded(`/terms/${props.location.search}`));
    this.props.dispatch(fetchThemesIfNeeded());
  }

  componentWillMount() {
    const tabCode = getParamFromUrl('tab');
    if (tabCode !== null) {
      if (tabCode === 'datasets') {
        ReactGA.modalview('/datasets');
        this.setState({
          showConcepts: false
        });
      } else if (tabCode === 'concepts') {
        ReactGA.modalview('/concepts');
        this.setState({
          showConcepts: true
        });
      }
    }
  }

  componentWillReceiveProps(nextProps) {
    const datasetQuery = `datasets/${nextProps.location.search}`;
    console.log("old props", JSON.stringify(this.props.location.search));
    console.log("next props", JSON.stringify(nextProps.location.search));
    //if(nextProps.location !== this.props.location) {
    if(nextProps.location.search !== this.props.location.search) {
      this.props.dispatch(fetchDatasetsIfNeeded(`/datasets/${nextProps.location.search}`));
      this.props.dispatch(fetchTermsIfNeeded(`/terms/${nextProps.location.search}`));
    }
  }

  handleSelectView(chosenView) {
    let tabUrl = getTabUrl(chosenView);
    tabUrl = removeParam('p', tabUrl); // remove this parameter when navigating between tabs.
    const nextUrl = `${location.pathname}${tabUrl}`;
    browserHistory.push(nextUrl);

    if (chosenView === 'datasets') {
      ReactGA.modalview('/datasets');
      this.setState({
        showConcepts: false
      });
    } else if (chosenView === 'concepts') {
      ReactGA.modalview('/concepts');
      this.setState({
        showConcepts: true
      });
    }
  }

  handleHistoryListen(history, location) {
    if(location.search.indexOf('lang=') === -1 && this.props.selectedLanguageCode && this.props.selectedLanguageCode !== "nb") {
      let nextUrl = "";
      if (location.search.indexOf('?') === -1) {
        nextUrl = `${location.search}?lang=${   this.props.selectedLanguageCode}`
      } else {
        nextUrl = `${location.search}&lang=${   this.props.selectedLanguageCode}`
      }
      history.push(nextUrl);
    }

    if (location.search.indexOf('tab=') === -1 && this.state.showConcepts) {
      let nextUrl = "";
      if (location.search.indexOf('?') === -1 && this.state.showConcepts) {
        nextUrl = `${location.search}?tab=concepts`
      } else {
        nextUrl = `${location.search}&tab=concepts`
      }
      history.push(nextUrl);
    }
  }

  handleSearchSubmit(event) {
    this.props.history.push(`?${qs.stringify(this.state.searchQuery, { skipNulls: true })}`);
  }

  handleSearchChange(event) {
    this.setState(
      {
        searchQuery: {
          ...this.state.searchQuery,
          q: event.target.value !== '' ? event.target.value : null
        }
      }
    );
  }

  handleDatasetFilterThemes(event) {
    const { theme } = this.state.searchQuery;
    if (event.target.checked) {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            theme:  addValue(theme, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
    else {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            theme: removeValue(theme, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
  }

  handleDatasetFilterAccessRights(event) {
    const { accessrights } = this.state.searchQuery;
    if (event.target.checked) {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            accessrights: addValue(accessrights, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
    else {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            accessrights: removeValue(accessrights, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
  }

  handleDatasetFilterPublisher(event) {
    const { publisher } = this.state.searchQuery;
    if (event.target.checked) {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            publisher: addValue(publisher, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
    else {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            publisher: removeValue(publisher, event.target.value)
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
  }

  handleDatasetSort(event) {
    let sortField = event.field;

    if (sortField === '_score') {
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            sortfield: undefined,
            sortdirection: undefined
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    } else {
      if (sortField === 'title') {
        sortField = sortField.concat('.').concat(localization.getLanguage())
      }
      this.setState(
        ({
          searchQuery: {
            ...this.state.searchQuery,
            sortfield: sortField,
            sortdirection: event.order
          }
        }),
        () => this.props.history.push(`?${qs.stringify(this.state.searchQuery)}`)
      );
    }
  }

  close() {
    this.setState({ showFilterModal: false });
  }

  open() {
    this.setState({ showFilterModal: true });
  }

  render() {
    const { selectedLanguageCode, datasetItems, isFetchingDatasets, termItems, isFetchingTerms, themesItems }  = this.props;
    const showDatasets = cx(
      {
        show: !this.state.showConcepts,
        hide: this.state.showConcepts
      }
    );
    const showConcepts = cx(
      {
        show: this.state.showConcepts,
        hide: !this.state.showConcepts
      }
    );

    /*
     <div className={showDatasets}>
     <ResultsDataset
     onHistoryListen={this.handleHistoryListen}
     onSelectView={this.handleSelectView}
     isSelected={!this.state.showConcepts}
     selectedLanguageCode={this.props.selectedLanguageCode}
     datasetItems={datasetItems}
     location={this.props.location}
     history={this.props.history}
     match={this.props.match}
     onSearch={this.handleSearch}
     onFilterTheme={this.handleDatasetFilterThemes}
     onFilterAccessRights={this.handleDatasetFilterAccessRights}
     onFilterPublisher={this.handleDatasetFilterPublisher}
     onSort={this.handleDatasetSort}
     searchQuery={this.state.searchQuery}
     themesItems={themesItems}
     showFilterModal={this.state.showFilterModal}
     closeFilterModal={this.close}
     />
     </div>

     <div className={showConcepts}>
     <ResultsConcepts
     onSelectView={this.handleSelectView}
     isSelected={this.state.showConcepts}
     onHistoryListen={this.handleHistoryListen}
     selectedLanguageCode={this.props.selectedLanguageCode}
     />
     </div>
     */
    return (
      <div className="container">
        <SearchBox
          onSearchSubmit={this.handleSearchSubmit}
          onSearchChange={this.handleSearchChange}
          searchQuery={this.state.searchQuery.q}
          countDatasets={(datasetItems && datasetItems.hits) ? datasetItems.hits.total : null}
          isFetchingDatasets={isFetchingDatasets}
          countTerms={(termItems && termItems.hits) ? termItems.hits.total : null}
          isFetchingTerms={isFetchingTerms}
          open={this.open}
        />
        <ResultsTabs
          onSelectView={this.handleSelectView}
          location={this.props.location}
          countDatasets={(datasetItems && datasetItems.hits) ? datasetItems.hits.total : null}
          countTerms={(termItems && termItems.hits) ? termItems.hits.total : null}
          selectedLanguageCode={selectedLanguageCode}
        />
        <Switch>
          <Route
            exact
            path="/"
            render={(props) =>
              (<ResultsDataset
                onHistoryListen={this.handleHistoryListen}
                selectedLanguageCode={this.props.selectedLanguageCode}
                datasetItems={datasetItems}
                onFilterTheme={this.handleDatasetFilterThemes}
                onFilterAccessRights={this.handleDatasetFilterAccessRights}
                onFilterPublisher={this.handleDatasetFilterPublisher}
                onSort={this.handleDatasetSort}
                searchQuery={this.state.searchQuery}
                themesItems={themesItems}
                showFilterModal={this.state.showFilterModal}
                closeFilterModal={this.close}
                {...props}
              />)
            }
          />
          <Route
            exact
            path="/concepts/:lang?"
            render={(props) =>
              (<ResultsConcepts
                onHistoryListen={this.handleHistoryListen}
                selectedLanguageCode={this.props.selectedLanguageCode}
                termItems={termItems}
                {...props}
              />)
            }
          />
        </Switch>

      </div>
    );
  }
}

SearchPage.defaultProps = {
  selectedLanguageCode: null
};

SearchPage.propTypes = {
  selectedLanguageCode: PropTypes.string
};

function mapStateToProps({ datasets, terms, themes }, ownProps) {

  const { datasetItems, isFetchingDatasets } = datasets || {
    datasetItems: null
  }

  const { termItems, isFetchingTerms } = terms || {
    termItems: null
  }

  const { themesItems, isFetchingThemes } = themes || {
    themesItems: null
  }

  return {
    datasetItems,
    isFetchingDatasets,
    termItems,
    isFetchingTerms,
    themesItems,
    isFetchingThemes
  }
}

export default connect(mapStateToProps)(SearchPage);
