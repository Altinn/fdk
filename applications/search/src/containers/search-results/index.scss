@import '../../assets/css/colors.scss';
/* ==========================================================================
   Author's custom styles
   ========================================================================== */

/*  SEARCHPAGE SEARCHBOX AND TEXT  */
.mb-60 {
  margin-bottom: 60px;
}

.fdk-search-flex {
  display: flex;
  align-items: flex-start;
}

.sk-hits-stats__info {
  color: $color1;
  font-size: 2rem;
  font-weight: 500;
}

/* SEARCHPAGE FILTERS */

/*
.filter--theme{
  .sk-panel__header {
    &:before {
      content: "\F02E";
      display: inline-block;
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      text-rendering: auto;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      margin-right: .5em;
    }
  }
}

.filter--publisher {
  .sk-panel__header {
    &:before {
      content: "\F0F7";
      display: inline-block;
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      text-rendering: auto;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      margin-right: .5em;
    }
  }
}

.filter--accessRight {
  .sk-panel__header {
    &:before {
      content: "\F084";
      display: inline-block;
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      text-rendering: auto;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      margin-right: .5em;
    }
  }
}
*/

/*
.search-filters .sk-item-list > input[type=checkbox] {
  position: absolute;
  left:-9999px;
}
.search-filters .sk-item-list > label {
  cursor: pointer;
}
.search-filters .sk-item-list > label:last-child {
  display: none;
}
.search-filters div.checkbox:nth-child(-n+5) {
  display: block;
}
.search-filters input[type=checkbox]:checked ~ div {
  display: block !important;
}
.search-filters input[type=checkbox]:checked ~ label {
  display: none !important;
}
.search-filters input[type=checkbox]:checked ~ label:last-child {
  display: inline-block !important;
}
.search-filters div.checkbox {
  display: none;
}
*/

/*  SEARCHPAGE PAGINATION  */

.sk-pagination-navigation .sk-toggle-option:first-child,
.sk-pagination-navigation .sk-toggle-option:last-child {
  flex: auto;
}

.sk-pagination-navigation {
  margin:10px auto;
}

.sk-pagination-select {
  margin: 10px auto;
  text-align:center;
}

.sk-toggle {
  display: flex;
  justify-content: center;
}
.sk-toggle-option {
  cursor: pointer;
  white-space: nowrap;
  font-size: 1.5rem;
  line-height: 1.5rem;
  padding: 11px 12px 10px 12px;
  margin: 0px 0px 6px 0px;
  //display: inline-block;
  //background-image: none;
  //text-shadow: none;
  text-align: center;
  text-decoration: underline;
  &:hover {
    background-color: $color5;
    text-decoration: none;
  }
}

.sk-toggle-option.sk-toggle__item.is-disabled {
  color: $color4;
}


.sk-toggle-option__text {
  &:before {
    display: inline-block;
    font: normal normal normal 14px/1 FontAwesome;
    font-size: inherit;
    text-rendering: auto;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
}

.sk-pagination-navigation .sk-toggle-option:first-child, .sk-pagination-navigation .sk-toggle-option:last-child {
  flex: auto;
}
.sk-toggle-option.is-disabled {
  cursor: initial;
  color: #ddd;
}
.sk-toggle-option:first-child {
  .sk-toggle-option__text {
    &:before {
      content: "\F060";
      display: inline-block;
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      text-rendering: auto;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      margin-right: .5em;
    }
  }
}
.sk-toggle-option:last-child {
  .sk-toggle-option__text {
    &:after {
      content: "\F061";
      display: inline-block;
      font: normal normal normal 14px/1 FontAwesome;
      font-size: inherit;
      text-rendering: auto;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      margin-left: .5em;
    }
  }
}
.sk-toggle-option.is-active {
  background-color: $color1;
  color: $color0;
  text-decoration: none;
}

/* SEARCHPAGE REFINEMENT FILTER */

.sk-panel {
  background-color: $color5;
  padding: 2rem;
  margin: .5em auto;
  border-radius: 5px;
}

.sk-panel__header {
  color: $color1;
  font-size: 1.875rem;
  line-height: 2.25rem;
  font-weight: 500;
  margin: 0px;
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {
  .sk-hits-stats__info {
    font-size: 2.25rem;
    line-height: 2.7rem;
  }
}

/* Large Devices, Wide Screens */
@media only screen and (min-width : 1200px) {

}
