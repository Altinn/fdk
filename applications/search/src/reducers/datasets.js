import _ from 'lodash';
import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../constants/ActionTypes';

export default function datasets(
  state = {
    isFetchingDatasets: false,
    datasetItems: null,
    publisherCountItems: null
  },
  action
) {
  switch (action.type) {
    case DATASETS_REQUEST: {
      return {
        ...state,
        isFetchingDatasets: true
      };
    }
    case DATASETS_SUCCESS: {
      const orgs = action.response.data.aggregations.orgPath.buckets;
      const flat = _(orgs).forEach(f => {
        const filteredOrgs = _(orgs)
          .filter(g => g.key.substring(0, g.key.lastIndexOf('/')) === f.key)
          .value();
        filteredOrgs.forEach(item => {
          const retVal = item;
          retVal.hasParent = true;
          return retVal;
        });
        const retVal = f;
        retVal.children = filteredOrgs;
        return retVal;
      });
      const resultArray = _(flat)
        .filter(f => !f.hasParent)
        .value();

      const objFromArray = action.response.data.aggregations.subjectsCount.buckets.reduce(
        (accumulator, current) => {
          accumulator[current.key] = current; // eslint-disable-line no-param-reassign
          return accumulator;
        },
        {}
      );
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: action.response.data,
        publisherCountItems: resultArray,
        subjectsCountItems: objFromArray
      };
    }
    case DATASETS_FAILURE: {
      return {
        ...state,
        isFetchingDatasets: false,
        datasetItems: null,
        publisherCountItems: null
      };
    }
    default:
      return state;
  }
}

export const getDatasetById = (datasets, id) => {
  if (datasets && datasets.datasetItems && datasets.datasetItems.hits) {
    return datasets.datasetItems.hits.hits.filter(
      dataset => dataset._source.id === id
    );
  }
  return null;
};
