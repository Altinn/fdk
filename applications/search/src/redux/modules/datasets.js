import _ from 'lodash';
import { normalizeAggregations } from '../../lib/normalizeAggregations';
import { fetchActions } from '../fetchActions';

export const DATASETS_REQUEST = 'DATASETS_REQUEST';
export const DATASETS_SUCCESS = 'DATASETS_SUCCESS';
export const DATASETS_FAILURE = 'DATASETS_FAILURE';

function shouldFetch(metaState, query) {
  const threshold = 60 * 100; // seconds
  return (
    !metaState ||
    (!metaState.isFetching &&
      ((metaState.lastFetch || 0) < Date.now() - threshold ||
        metaState.cachedQuery !== query))
  );
}

export function fetchDatasetsIfNeededAction(query) {
  return (dispatch, getState) =>
    shouldFetch(_.get(getState(), ['datasets', 'meta']), query) &&
    dispatch(
      fetchActions(`/datasets?${query}`, [
        DATASETS_REQUEST,
        { type: DATASETS_SUCCESS, meta: { query } },
        DATASETS_FAILURE
      ])
    );
}

const initialState = {};

export function datasets(state = initialState, action) {
  switch (action.type) {
    case DATASETS_REQUEST:
      return {
        ...state,
        meta: {
          isFetching: true,
          lastFetch: null,
          cachedQuery: null
        }
      };
    case DATASETS_SUCCESS:
      return {
        ...state,
        datasetItems: _.get(action.payload, ['hits', 'hits']),
        datasetAggregations: _.get(normalizeAggregations(action.payload), [
          'aggregations'
        ]),
        datasetTotal: _.get(action.payload, ['hits', 'total']),
        meta: {
          isFetching: false,
          lastFetch: Date.now(),
          cachedQuery: action.meta.query
        }
      };
    case DATASETS_FAILURE:
      return {
        ...state,
        datasetItems: null,
        datasetAggregations: null,
        datasetTotal: null,
        meta: {
          isFetching: false,
          lastFetch: null,
          cachedQuery: null
        }
      };
    default:
      return state;
  }
}
