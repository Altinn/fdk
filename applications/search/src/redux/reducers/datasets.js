import _ from 'lodash';
import {
  DATASETS_REQUEST,
  DATASETS_SUCCESS,
  DATASETS_FAILURE
} from '../ActionTypes';

export default function datasets(
  state = {
    isFetching: false,
    datasetItems: null,
    publisherCountItems: null
  },
  action
) {
  switch (action.type) {
    case DATASETS_REQUEST: {
      return {
        ...state,
        isFetching: true
      };
    }
    case DATASETS_SUCCESS: {
      const orgs = action.payload.aggregations.orgPath.buckets;
      const flat = _(orgs).forEach(f => {
        const filteredOrgs = _(orgs)
          .filter(g => g.key.substring(0, g.key.lastIndexOf('/')) === f.key)
          .value();
        filteredOrgs.forEach(item => {
          const retVal = item;
          retVal.hasParent = true;
          return retVal;
        });
        const retVal = f;
        retVal.children = filteredOrgs;
        return retVal;
      });
      const resultArray = _(flat)
        .filter(f => !f.hasParent)
        .value();

      const objFromArray = action.payload.aggregations.subjectsCount.buckets.reduce(
        (accumulator, current) => {
          accumulator[current.key] = current; // eslint-disable-line no-param-reassign
          return accumulator;
        },
        {}
      );
      return {
        ...state,
        isFetching: false,
        datasetItems: action.payload,
        publisherCountItems: resultArray,
        subjectsCountItems: objFromArray
      };
    }
    case DATASETS_FAILURE: {
      return {
        ...state,
        isFetching: false,
        datasetItems: null,
        publisherCountItems: null
      };
    }
    default:
      return state;
  }
}

export const getDatasetById = (datasets, id) => {
  if (datasets && datasets.datasetItems && datasets.datasetItems.hits) {
    return datasets.datasetItems.hits.hits.filter(
      dataset => dataset._source.id === id
    );
  }
  return null;
};
