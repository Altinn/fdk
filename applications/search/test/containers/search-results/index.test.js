import React from 'react';
import { shallow } from 'enzyme';
import { SearchPage } from '../../../src/containers/search-results';
import datasets from '../../fixtures/datasets';

let defaultProps, wrapper;

beforeEach(() => {
  defaultProps = {
    location: {"pathname":"/","search":"","hash":"","key":"lj1xo0"}
  };
  wrapper = shallow(
    <SearchPage
      {...defaultProps}
    />
  )
});

test('should render SearchPage correctly with no datasetItems', () => {
  expect(wrapper).toMatchSnapshot();
})



