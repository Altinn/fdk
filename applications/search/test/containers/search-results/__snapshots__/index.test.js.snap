// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchPage correctly with no datasetItems 1`] = `undefined`;
